### ISSUE TITLE
New Rule: Feature Request: Linter rule for undefined variables in template literals

### ISSUE THREAD

**Issue by john1625b** [eslint/eslint#17855]
### Rule details

Warn when potentially undefined is used in string template

### Related ECMAScript feature

ES14

### What type of rule is this?

Warns about a potential problem

### Example code

```js
let res;
let result = `${res} bar`;  // result becomes "undefined bar"
```

### Why should this rule be in the core instead of a plugin?

Hello,

I’ve encountered a situation in my JavaScript code where I’m using template literals for string interpolation, and there’s a chance that some variables could be undefined. When an undefined variable is used in a template literal, it gets converted to the string "undefined", which can lead to unexpected results.

 it would be great if there was a linter rule that could catch potentially undefined variables in template literals. This would help prevent such issues from occurring in the first place.

### Participation

- [ ] I am willing to submit a pull request to implement this rule.

### Additional comments

The user can then be signaled  to use a short-circuit evaluation:

```
let foo;
let result = `${foo || ''} bar`;  // result becomes " bar"
```

**Comment by aladdin-add** [comment]
Template literals was introduced in es2015. Per our [rules policies](https://eslint.org/blog/2020/05/changes-to-rules-policies), we can only accept new rules if they relate to new ECMAScript features that reached stage 4 in the preceding 12 months, so this proposal doesn't qualify. You can always make a custom rule and publish a plugin.
